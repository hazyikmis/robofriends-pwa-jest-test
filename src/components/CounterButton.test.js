import { shallow } from "enzyme";
import React from "react";
import CounterButton from "./CounterButton";

it("snapshot match testing: CounterButton component", () => {
  const mockColor = "red";
  expect(
    shallow(<CounterButton color={mockColor} />).debug()
  ).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  wrapper.find('[id="counter"]').simulate("click");
  //wrapper.find('[id="counter"]').simulate('click');
  wrapper.find("button").simulate("click"); //same button clicked!!!
  //wrapper.find('button').simulate('click');
  expect(wrapper.state()).toEqual({ count: 3 });
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 4 });
  wrapper.find('[id="counter"]').simulate("keypress");
  expect(wrapper.state()).toEqual({ count: 4 });
});

it("props.color check", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);
  //expect(wrapper.props()).toEqual({ color: "red" }); //this DOES NOT WORKS, because %99.9 props have many objects, not only color
  expect(wrapper.props().color).toEqual(mockColor);
});
