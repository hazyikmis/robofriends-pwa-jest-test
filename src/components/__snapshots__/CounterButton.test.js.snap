// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot match testing: CounterButton component 1`] = `
"<button id=\\"counter\\" color=\\"red\\" onClick={[Function: onClick]}>
  Count: 
  1
</button>"
`;
